### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by vvoody-**
- removed steevanb/doctrine-read-only-hydrator
- created FileEntity interface
- FileListener do not need entityClassName and EntityManager
- createdAt removed (its not purpose of this library, can be added in userland side)
- need of salt was replaced with random string generation
- FileListener implements EventSubscriber from Doctrine not Kdyby
- code style reformating
- declare strict types
- fixed removing multiple files in one flush
- refactoring of some method names
- removed usage of smart object
- updated readme

### CODE REVIEW

File: src/Entities/FileEntity.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace ADT\Files\Entities;
+
+interface FileEntity
```
Comment (thorewi) [inline src/Entities/FileEntity.php]: IFileEntity

File: src/Entities/FileTrait.php [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 
 use ADT\Files\Helpers;
 use Doctrine\ORM\Mapping as ORM;
 
-trait FileTrait 
+/**
+ * Trait FileTrait
+ * @package ADT\Files\Entities
+ */
+trait FileTrait
```
Comment (thorewi) [inline src/Entities/FileTrait.php]: TFileEntity

File: src/Entities/FileTrait.php [hunk 3]
```diff
@@ -53,70 +51,51 @@ trait FileTrait
...
-	}
-
-	public function getPath()
+	public function getPath(): string
 	{
 		return $this->path . '/' . $this->filename;
 	}
 
-	public function setPath(string $path)
+	public function setBaseDirectoryPath(string $path): void
```
Comment (thorewi) [inline src/Entities/FileTrait.php]: vratit vsude fluent

File: src/Listeners/FileListener.php [hunk 4]
```diff
@@ -82,53 +83,52 @@ public function postPersist(LifecycleEventArgs $args)
...
+
+
+		// smazane file entity si ulozime a pak je v post flush pouzijeme pro smazani jejich soboru
+		// post remove nelze pouzit lebo sa ne vzdy zavola (treba pri smazani pres orphanremoval)
+		// https://github.com/doctrine/orm/issues/6256
+		$this->filesToDelete[] = $entity;
 	}
 
-	public function postRemove(\Doctrine\ORM\Event\LifecycleEventArgs $eventArgs)
+	public function postFlush()
```
Comment (thorewi) [inline src/Listeners/FileListener.php:119]: mozna v tomto byl ten problem, ze ja jsem to delal v postRemove a proto se mi ta iniciovana entita znovu hodila do db... ty kdyz to delas v postFlush tak se to mozna diky tomu nedeje...

Comment (vvoody-) [reply src/Listeners/FileListener.php:119]: Puvodne sem to mel v postRemove a taky to slo v pohode.

File: src/Entities/FileEntity.php (hunk 1 above)
Comment (vvoody-) [reply src/Entities/FileEntity.php]: hotovo

File: src/Entities/FileTrait.php (hunk 2 above)
Comment (vvoody-) [reply src/Entities/FileTrait.php]: hotovo

File: src/Entities/FileTrait.php (hunk 3 above)
Comment (vvoody-) [reply src/Entities/FileTrait.php]: hotovo

**Pull request merged**